probe-rs/src/probe.rs

```diff
@@ -538,6 +548,19 @@ pub trait DebugProbe: Send + fmt::Debug {
     ///
     fn set_speed(&mut self, speed_khz: u32) -> Result<u32, DebugProbeError>;
 
+    /// Set the JTAG scan chain information for the target under debug.
+    ///
+    /// This allows the probe to know which TAPs are in the scan chain and their
+    /// position and IR lengths.
+    ///
+    /// If the scan chain is provided, and the selected protocol is JTAG, the
```

Comment:
This might be misleading because depending on when one calls `set_scan_chain`, the call might not have any effect because the change will not be applied. Maybe we should clarify that bit?